src/main/java/com/airlines/dao/impl/AirplaneDaoImpl.java

```diff
@@ -78,60 +81,95 @@ public AirplaneDaoImpl(DataSource dataSource) {
                         .build());
             }
         } catch (SQLException e) {
-            throw new DaoOperationException("Couldn't save product ", e);
+            throw new DaoOperationException("Couldn't find an airplane ", e);
         }
         return airplaneArray;
     }
 
     @Override
-    public Airplane findOne(String codeName) {
-        Airplane airplane = null;
+    public Optional<Airplane> findByCodeName(String codeName) {
         try (Connection connection = dataSource.getConnection();
-             PreparedStatement selectByIdStatement = connection.prepareStatement(SELECT_BY_CODE_NAME_SQL);) {
-            selectByIdStatement.setString(1, codeName);
-            ResultSet resultSet = selectByIdStatement.executeQuery();
+             PreparedStatement selectByCodeNameStatement = connection.prepareStatement(SELECT_BY_CODE_NAME_QUERY);) {
+            selectByCodeNameStatement.setString(1, codeName);
+            ResultSet resultSet = selectByCodeNameStatement.executeQuery();
             if (resultSet.next()) {
-                airplane = Airplane.builder()
+                return Optional.of(Airplane.builder()
                         .withId(resultSet.getLong("id"))
                         .withCodeName(resultSet.getString("name"))
                         .withModel(resultSet.getString("model"))
                         .withManufactureDate(resultSet.getDate("manufactureDate").toLocalDate())
                         .withCapacity(resultSet.getInt("capacity"))
                         .withFlightRange(resultSet.getInt("flightRange"))
                         .withCrew((Crew) resultSet.getObject("crew"))
```

Comment:
It's look like we have make change to domain model.
resultSet.getObject("crew") will return to us Crew id - Long object.

For turning it into a Crew there is no other way as to make select crew by id with using CrewDaoImpl.

If you not understand this moment , just let me know and ask for clarifying.

So, rewrite Airplane entity and change field Crew crew to Long crewId
Also review CrewMemberDaoImpl and AirplaneDaoImpl, maybe we have change there some logic as well